### DISCUSSION TITLE
Importing Google photos from extracted takeout

### DISCUSSION THREAD

**Discussion by tsemachh** [simulot/immich-go#63] [Q&A] [answered]
Hi
I am trying to import google photos from extracted takeout(Needed to convert the BMP files to JPEG)
I am running:
 ./immich-go.exe -server=http://immich:2283 -key=<MYKEY> upload -create-albums -google-photos "E:\JPG61_66\takeout-20231019T014649Z-061\Takeout\Google Photos"
I am getting next error:
←[32mUploading "October 9 2023 - Toronto/frames-4568.jpg"...←[31mError: AssetUpload, POST, http://immich:2283/api/asset/upload, 400 Bad Request
Bad Request

Am I doing it right ? Does Google photos extraction need to have the takeout files , or I can run it against the extracted directory ?

**Comment by simulot** [comment]
Have you tried to drop the jpeg in the immich page?

**Reply by tsemachh** [reply]
Yes the file uplodas O.K maybe the issue is with the json:
{
  "title": "bla bla bla.bmp",
  "description": "",
  "imageViews": "1",
  "creationTime": {
    "timestamp": "1689310435",
    "formatted": "Jul 14, 2023, 4:53:55 AM UTC"
  },
  "photoTakenTime": {
    "timestamp": "1689308119",
    "formatted": "Jul 14, 2023, 4:15:19 AM UTC"
  },
  "geoData": {
    "latitude": 0.0,
    "longitude": 0.0,
    "altitude": 0.0,
    "latitudeSpan": 0.0,
    "longitudeSpan": 0.0
  },
  "geoDataExif": {
    "latitude": 0.0,
    "longitude": 0.0,
    "altitude": 0.0,
    "latitudeSpan": 0.0,
    "longitudeSpan": 0.0
  },
  "people": [{
    "name": "Hadad Shalom"
  }],
  "url": "https://photos.google.com/photo/AF1QipNkQ0d4hrVQd-NH5bDLAxal2rRqZeqXZPNTrH6T",
  "googlePhotosOrigin": {
    "webUpload": {
      "computerUpload": {
      }
    }
  }
}

When I took the entire archive some succeeded to upload but many failed with Bad Request

**Reply by simulot** [reply]
Nothing strange here

**Comment by tsemachh** [comment]
Maybe I can use fiddler tool to check the network , will update if I see something

**Reply by simulot** [reply]
Before going to fiddler, you may try to dump the api call
```
./immich-go.exe -server=http://immich:2283/ -key=xxxx -api-trace=true -log-level=debug upload -create-albums -google-photos "E:\JPG61_66\takeout-20231019T014649Z-061\Takeout\Google Photos"
``

**Reply by tsemachh** [reply]
Here's the dump:
GET http://immich:2283/api/server-info/ping
Accept [application/json]
X-Api-Key [immichapikey]
←[32mServer status: OK←[39m
--------------------
GET http://immich:2283/api/user/me
X-Api-Key [immichapikey]
Accept [application/json]
←[37mConnected, user: me@me.com←[39m
←[32mAsk for server's assets...←[39m
--------------------
GET http://immich:2283/api/asset
Accept [application/json]
X-Api-Key [immichapikey]
←[32m972 asset(s) received←[39m
←[32mBrowsing google take out archive...←[32mDone.←[39m
←[37mTsemach Me Flikr 2-1108.bmp: This a new asset, upload it.←[39m
←[32mUploading "Tsemach Me Flikr 2-1108.jpg"...--------------------
POST http://immich:2283/api/asset/upload
Accept [application/json]
Content-Type [multipart/form-data; boundary=fc16640d0b72170ee3cfabc1e170c7178a8fa5b9f46ee0b3f3806256b93e]
X-Api-Key [immichapikey]
--fc16640d0b72170ee3cfabc1e170c7178a8fa5b9f46ee0b3f3806256b93e
Content-Disposition: form-data; name="deviceAssetId"

Tsemach Me Flikr 2-1108.bmp-164180
--fc16640d0b72170ee3cfabc1e170c7178a8fa5b9f46ee0b3f3806256b93e
Content-Disposition: form-data; name="deviceId"

TsemachD-LT
--fc16640d0b72170ee3cfabc1e170c7178a8fa5b9f46ee0b3f3806256b93e
Content-Disposition: form-data; name="assetType"

IMAGE
--fc16640d0b72170ee3cfabc1e170c7178a8fa5b9f46ee0b3f3806256b93e
Content-Disposition: form-data; name="fileCreatedAt"

2023-07-14T07:15:19+03:00
--fc16640d0b72170ee3cfabc1e170c7178a8fa5b9f46ee0b3f3806256b93e
Content-Disposition: form-data; name="fileModifiedAt"

2023-07-14T07:15:19+03:00
--fc16640d0b72170ee3cfabc1e170c7178a8fa5b9f46ee0b3f3806256b93e
Content-Disposition: form-data; name="isFavorite"

false
--fc16640d0b72170ee3cfabc1e170c7178a8fa5b9f46ee0b3f3806256b93e
Content-Disposition: form-data; name="fileExtension"

.jpg
--fc16640d0b72170ee3cfabc1e170c7178a8fa5b9f46ee0b3f3806256b93e
Content-Disposition: form-data; name="duration"

00:00:00.000000
--fc16640d0b72170ee3cfabc1e170c7178a8fa5b9f46ee0b3f3806256b93e
Content-Disposition: form-data; name="isReadOnly"

false
--fc16640d0b72170ee3cfabc1e170c7178a8fa5b9f46ee0b3f3806256b93e
Content-Disposition: form-data; name="assetData"; filename="Tsemach Me Flikr 2-1108.bmp"
Content-Type: image/jpeg

���� ►JFIF ☺☺☺ H H  ��☼CExif  I...
--- BODY ---
Error: AssetUpload, POST, http://immich:2283/api/asset/upload, 400 Bad Request

**Reply by simulot** [reply]
thanks. 404 message comes from immich to say it doesn't like what we request. But it doesn't give details.

Could you take the file `Tsemach Me Flikr 2-1108.bmp` apart, and just upload it? 
Wait... why the file extension is .BMP, when the content type is image/jpeg ?

**Reply by tsemachh** [reply]
It's a BMP I converted to JPEG , maybe I need to change also in the JSON the title to jpg, will give it a try

**Reply by tsemachh** [reply, marked as answer]
This fixed the problem - thank you will convert the json files name to fit new name and will run full upload again
